core/src/test/java/io/aiven/klaw/service/UsersTeamsControllerServiceTest.java

```diff
@@ -381,7 +458,111 @@ void addNewUserInvalidPatternFailure() throws KlawException {
   }
 
   @Test
-  void addNewTeam() {}
+  void addNewTeamSuccess() throws KlawException {
+    int tenantId = 101;
+    String userName = "testuser";
+    String existingTeamName = "existingTeamName";
+    TeamModel teamModel = utilMethods.getTeamModelMock();
+
+    when(mailService.getUserName(any())).thenReturn(userName);
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(tenantId);
+    when(manageDatabase.getTeamNamesForTenant(tenantId)).thenReturn(List.of(existingTeamName));
+    when(handleDbRequests.addNewTeam(teamCaptor.capture())).thenReturn(ApiResultStatus.SUCCESS.value);
+
+    ApiResponse apiResponse = usersTeamsControllerService.addNewTeam(teamModel, true);
+    Team capturedTeam = teamCaptor.getValue();
+
+    assertThat(capturedTeam.getRequestTopicsEnvs()).isEqualTo(
+            String.join(",", teamModel.getEnvList().toArray(new String[0])));
+    assertThat(capturedTeam.getTeamname()).isEqualTo(teamModel.getTeamname());
+    assertThat(capturedTeam.getTeamphone()).isEqualTo(teamModel.getTeamphone());
+    assertThat(capturedTeam.getContactperson()).isEqualTo(teamModel.getContactperson());
+    assertThat(capturedTeam.getTenantId()).isEqualTo(tenantId);
+    assertThat(capturedTeam.getTeamId()).isEqualTo(teamModel.getTeamId());
+
+    assertThat(apiResponse.isSuccess()).isTrue();
+    assertThat(apiResponse.getMessage()).isEqualTo(ApiResultStatus.SUCCESS.value);
+    verify(commonUtilsService).updateMetadata(tenantId, EntityType.TEAM, MetadataOperationType.CREATE, null);
+  }
+
+  @Test
+  void addNewTeamWithApiFailureFromDB() throws KlawException {
```

Comment:
Sorry last comment, I missed this in my initial review, we should have our tests as public void as per convention for the other tests in the class.

Reply:
I have added public to all tests, including the tests that I did not write, I also used the `TEST_TENANT_ID `in the tests that I did not write such as `getAllTeamsSUOnly`, but I left the `teamId `untouched, as that would have caused some cascading changes in other files.

Reply:
Thanks a million, appreciate that!